monkey/infection_monkey/network/network_scanner.py

```diff
@@ -1,28 +1,50 @@
 import time
+import logging
 
-from common.network.network_range import *
+from common.network.network_range import NetworkRange
 from infection_monkey.config import WormConfiguration
+from infection_monkey.model.host import generate_victims_from_range
 from infection_monkey.network.info import local_ips, get_interfaces_ranges
-from infection_monkey.model import VictimHost
 from infection_monkey.network import TcpScanner, PingScanner
+from infection_monkey.utils import is_windows_os
 
-__author__ = 'itamar'
+if is_windows_os():
+    from multiprocessing.dummy import Pool
+else:
+    from multiprocessing import Pool
 
 LOG = logging.getLogger(__name__)
 
-SCAN_DELAY = 0
+ITERATION_BLOCK_SIZE = 5
+
+
+def generate_victims(net_ranges, chunk_size):
```

Comment:
Also move this method to the 'VictimHostGenerator'. Make this generator have `net_ranges` prop and `bad_ips` prop. So in total generator class will contain 4 methods: `__init__( net_ranges, bad_ips )`, this method: `get_n_victims( n )`, `_generate_victims_from_range( net_range )` and `is_ip_scannable( ip )`. As it is now `get_n_victims( n )` would call `_generate_victims_from_range( net_range )` but it would also call `is_ip_scannable( ip )` to know if we should append victim to the chunk or not.

Reply:
VictimHostGenerator will only be used by the NetworkScanner and will basically have all the generation logic, leaving the scanner only to scan?

Reply:
Yes. It's more readable if NetworkScanner scans the network and VictimHostGenerator generates VictimHost objects from whatever compared to having factory methods "sprinkled" around in different files.